cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1777,13 +1777,28 @@ macro(build_thrift)
     set(THRIFT_DEPENDENCIES ${THRIFT_DEPENDENCIES} boost_ep)
   endif()
 
+  set(THRIFT_PATCH_COMMAND)
+  if(CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0)
```

Comment:
Why make this conditional? The patch should always be valid anyway.

Reply:
that is fair, this has been merged without any conditional: https://github.com/apache/thrift/commit/947ad66940cfbadd9b24ba31d892dfc1142dd330

Reply:
If we always use `PATCH_COMMAND`, `patch` command is required in a build machine. We want to avoid it. For example, `patch` command doesn't exist in most Windows machine.

Reply:
Ah, right. But the condition seems overly specific in any case.

Reply:
Hmm. I don't think that the current condition has a problem because GCC 15 is the only known problematic compiler for now. And this is a temporary condition. We will remove this patch when a new Apache Thrift is released.

What is your real concern?

Reply:
I was just surprised, no specific concern. Especially if Thrift has incorporated the patch anyway.